store/tikv/scan.go

```diff
@@ -172,6 +179,18 @@ func (s *Scanner) getData(bo *Backoffer) error {
 				NotFillCache: s.snapshot.notFillCache,
 			},
 		}
+
+		// when endkey is beyond current region, stop at current boundary.
+		if s.reverse {
+			if len(s.endKey) > 0 && len(loc.StartKey) > 0 && bytes.Compare(loc.StartKey, s.endKey) < 0 {
```

Comment:
Should it be `bytes.Compare(loc.StartKey, s.endKey) > 0`?

Reply:
I don't think so, this situation is location is contained in [s.StartKey, s.EndKey] or [s.EndKey, s.StartKey].

@disksing

Reply:
> I don't think so, this situation is location is contained in [s.StartKey, s.EndKey] or [s.EndKey, s.StartKey].
> 
> @disksing

oh I made a mistake, this should be `>` here.

Reply:
fixed b3a9a61b7